server/test/controllers/admin/AdminProgramBlocksControllerTest.java

```diff
@@ -99,6 +110,74 @@ public void create_withProgram_addsRepeatedBlock() {
     assertThat(program.getProgramDefinition().blockDefinitions()).hasSize(3);
   }
 
+  @Test
+  public void view_throwsError_ifCalledOnNoneActiveProgram() throws Exception {
+    Request request = addCSRFToken(Helpers.fakeRequest()).build();
+    Program program = ProgramBuilder.newDraftProgram("test program").build();
+
+    assertThatThrownBy(() -> controller.view(request, program.id, 1L))
+        .isInstanceOf(NotViewableException.class);
+  }
+
+  @Test
+  public void view_withInvalidProgram_notFound() {
+    Request request = fakeRequest().build();
+    assertThatThrownBy(() -> controller.view(request, 1L, 1L))
+        .isInstanceOf(NotViewableException.class);
+  }
+
+  @Test
+  public void view_withInvalidBlock_notFound() {
+    Program program = ProgramBuilder.newActiveProgram().build();
+    Request request = fakeRequest().build();
+    Result result = controller.view(request, program.id, 2L);
+
+    assertThat(result.status()).isEqualTo(NOT_FOUND);
+  }
+
+  @Test
+  public void view_withProgram_OK()
+      throws UnsupportedQuestionTypeException, InvalidUpdateException {
+    Program program =
+        ProgramBuilder.newActiveProgram("Public name", "Public description")
+            // Override only admin name and description to distinguish from applicant-visible
+            // name/description.
+            .withName("Admin name")
+            .withDescription("Admin description")
+            .build();
+    Question appName = testQuestionBank.applicantName();
```

Comment:
**suggestion (non-blocking):** Use applicantName, app is too ambiguous with application